tidy3d/plugins/microwave/path_integrals.py

```diff
@@ -0,0 +1,299 @@
+        coords_interp = np.array([min_bound])
+        coords_interp = np.concatenate((coords_interp, coordinates))
+        coords_interp = np.concatenate((coords_interp, [max_bound]))
+        coords_interp = {coord: coords_interp}
+
+        # Use extrapolation for the 2 additional endpoints, unless there is only a single sample point
+        method = "linear"
+        if len(coordinates) == 1 and self.extrapolate_to_endpoints:
+            method = "nearest"
+        scalar_field = scalar_field.interp(
+            coords_interp, method=method, kwargs={"fill_value": "extrapolate"}
+        )
+        return scalar_field.integrate(coord=coord)
+
+    def _get_field_along_path(
+        self, scalar_field: AbstractSpatialDataArray
+    ) -> AbstractSpatialDataArray:
+        """Returns a selection of the input `scalar_field` ready for integration."""
+        axis1 = self.remaining_axes[0]
+        axis2 = self.remaining_axes[1]
+        coord1 = "xyz"[axis1]
+        coord2 = "xyz"[axis2]
+
+        if self.snap_path_to_grid:
+            # Coordinates that are not integrated
+            remaining_coords = {
+                coord1: self.center[axis1],
+                coord2: self.center[axis2],
+            }
+            # Select field nearest to center of integration line
+            scalar_field = scalar_field.sel(
+                remaining_coords,
+                method="nearest",
+                drop=False,
+            )
+        else:
+            # Try to interpolate unless there is only a single coordinate
+            coord1dict = {coord1: self.center[axis1]}
+            if scalar_field.sizes[coord1] == 1:
+                scalar_field = scalar_field.sel(coord1dict, method="nearest")
+            else:
+                scalar_field = scalar_field.interp(
+                    coord1dict, method="linear", kwargs={"bounds_error": True}
+                )
+            coord2dict = {coord2: self.center[axis2]}
+            if scalar_field.sizes[coord2] == 1:
+                scalar_field = scalar_field.sel(coord2dict, method="nearest")
+            else:
+                scalar_field = scalar_field.interp(
+                    coord2dict, method="linear", kwargs={"bounds_error": True}
+                )
+
+        return scalar_field
+
+    @cached_property
+    def main_axis(self) -> Axis:
+        """Axis for performing integration."""
+        val = next((index for index, value in enumerate(self.size) if value != 0), None)
+        return val
+
+
+class VoltageIntegralAA(AxisAlignedPathIntegral):
+    """Class for computing the voltage between two points defined by an axis-aligned line."""
+
+    sign: Direction = pd.Field(
+        "+",
+        title="Direction of path integral.",
+        description="Positive indicates V=Vb-Va where position b has a larger coordinate along the axis of integration.",
+    )
+
+    def compute_voltage(self, em_field: FieldDataset | ModeSolverDataset | FieldTimeDataset):
```

Comment:
I might be wrong but `|` might not work in some versions of python, so might want to double check the tests and `ruff`. It's also possible this doesn't render well in our docs, so just be aware of that. 

One other option is to define
```
EMFieldTypes = Union[FieldDataset | ModeSolverDataset | FieldTimeDataset]
```
at the top of the file and use it for type annotations.